### PR TITLE
Fixed issue 146: Added null checks

### LINKED ISSUE #146: Shader constructor doesn't handle null streams properly
This is an old issue that has been re-introduced in commit bab446fcf46396a22aafbb156a1fc04b0217b24e when geometry shader support was added (see #127 and #129).

A null check is performed before each `StreamAdaptor` is instantiated, but the adaptors' `InputStreamPtr` member is being accessed without checking again, which throws a NullReferenceException if any of the passed streams are null.

*Graphics/Shader.cs, line 87*
```C#
CPointer = sfShader_createFromStream(
  vertexAdaptor.InputStreamPtr,
  geometryAdaptor.InputStreamPtr,
  fragmentAdaptor.InputStreamPtr);
```

*Possible solutions*
```C#
CPointer = sfShader_createFromStream(
  vertexAdaptor?.InputStreamPtr ?? IntPtr.Zero,
  geometryAdaptor?.InputStreamPtr ?? IntPtr.Zero,
  fragmentAdaptor?.InputStreamPtr ?? IntPtr.Zero);
```
```C#
CPointer = sfShader_createFromStream(
  vertexAdaptor != null ? vertexAdaptor.InputStreamPtr : IntPtr.Zero,
  geometryAdaptor != null ? geometryAdaptor.InputStreamPtr : IntPtr.Zero,
  fragmentAdaptor != null ? fragmentAdaptor.InputStreamPtr : IntPtr.Zero);
```

I'm not sure whether the use of null conditional and coalescing operators in SFM.Net sources is allowed since they require C# 6.0 or higher, so I'll use the slightly longer syntax.

### PR DESCRIPTION
**Pull request by AtomCrafty**
Fixed #146 by adding null checks in the Shader(Stream, Stream, Stream) constructor.

### CODE REVIEW

**Comment by dabbertorres** [comment]
Good catch, thanks.

Looks good to me. Can we merge this? (@LaurentGomila, @eXpl0it3r, @SFML)

File: src/Graphics/Shader.cs [hunk 1]
```diff
@@ -81,10 +81,12 @@ public class CurrentTextureType { }
...
-                                geometryAdaptor = geometryShaderStream != null ? new StreamAdaptor(geometryShaderStream) : null,
-                                fragmentAdaptor = fragmentShaderStream != null ? new StreamAdaptor(fragmentShaderStream) : null)
+                                 geometryAdaptor = geometryShaderStream != null ? new StreamAdaptor(geometryShaderStream) : null,
+                                 fragmentAdaptor = fragmentShaderStream != null ? new StreamAdaptor(fragmentShaderStream) : null)
             {
-                CPointer = sfShader_createFromStream(vertexAdaptor.InputStreamPtr, geometryAdaptor.InputStreamPtr, fragmentAdaptor.InputStreamPtr);
+                CPointer = sfShader_createFromStream(vertexAdaptor != null ? vertexAdaptor.InputStreamPtr : IntPtr.Zero,
+                                                     geometryAdaptor != null ? geometryAdaptor.InputStreamPtr : IntPtr.Zero,
+	                                                 fragmentAdaptor != null ? fragmentAdaptor.InputStreamPtr : IntPtr.Zero);
             }
```
Comment (LaurentGomila) [inline src/Graphics/Shader.cs]: Minor alignment issue there

**Comment by AtomCrafty** [comment]
Whoops, looks like I forgot to get rid of a tab there ^^

**Comment by LaurentGomila** [comment]
Thanks :)

**Pull request merged**